### PR TITLE
feat: `func config volumes` - interactive prompt

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

Interactive prompt for managing function configuration - this PR adds Volumes (Envs will be in a separate PR).

`func config` - starts interactive prompt, wizard that let's you select resource and the requested operation

`func config volumes` - list configured Volumes

`func config volumes add` - starts interactive prompt, wizard that let's you add Volume mount to the configuration. It does a lookup for existing Secrets and ConfigMaps in the cluster and namespace.

`func config volumes remove` - starts interactive prompt, wizard that let's you remove Volume mount from the configuration.

Relates: #362

### CODE REVIEW

**Comment by zroubalik** [comment]
https://user-images.githubusercontent.com/726523/122253424-71090980-cecc-11eb-8ebd-48958528225c.mp4

**Review by lkingland — APPROVED** [review]
Nice.  Looking forward to using this flow elsewhere.  Like Lance said, this is a strong example of how func doesn't fit entirely into a plugin box.

File: cmd/config.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// CLI Configuration (parameters)
+// ------------------------------
+
+type configCmdConfig struct {
+	Name    string
+	Path    string
+	Verbose bool
+}
+
+func newConfigCmdConfig(args []string) configCmdConfig {
```
Comment (lance) [inline cmd/config.go:98]: Dangerously close to `AbstractConfigServletInitializerConfig` :rofl:

File: cmd/config_volumes.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		return
+	},
+}
+
+var configVolumesAddCmd = &cobra.Command{
+	Use:   "add",
+	Short: "Add volume to the function configuration",
+	Long: `Add volume to the function configuration
+
+Interactive prompt that allows adding Secrets and ConfigMaps as Volume mounts to the function project
```
Comment (lance) [inline cmd/config_volumes.go:52]: Nit
```suggestion
Interactive prompt to add Secrets and ConfigMaps as Volume mounts to the function project
```

File: cmd/config_volumes.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+		return runAddVolumesPrompt(cmd.Context(), function)
+	},
+}
+
+var configVolumesRemoveCmd = &cobra.Command{
+	Use:   "remove",
+	Short: "Remove volume from the function configuration",
+	Long: `Remove volume from the function configuration
+
+	Interactive prompt that allows removing Volume mounts from the configuration of the function project
```
Comment (lance) [inline cmd/config_volumes.go:73]:
```suggestion
	Interactive prompt to remove Volume mounts from the function project
```

`add` long description uses "as Volume mounts to the function project". But this says "removing Volume mounts from the configuration of the function project" which appears to have a different meaning and it's unclear if `add` would affect the cluster immediately or not. When we say "function project" we should always mean the configuration, so in this case I think using the words 'from the configuration" is probably redundant and could be confusing.

**Review by lance — APPROVED** [review]
I like it!

**Comment by lance** [comment]
> Nice. Looking forward to using this flow elsewhere. Like Lance said, this is a strong example of how func doesn't fit entirely into a plugin box.

Yes! I would like to update https://github.com/boson-project/func/pull/373 to use this flow for labels.

File: cmd/config.go (hunk 1 above)
Comment (zroubalik) [reply cmd/config.go:98]: Haha right 😅 I just wanted to follow the existing patterns for naming...

File: cmd/config_volumes.go (hunk 2 above)
Comment (zroubalik) [reply cmd/config_volumes.go:52]: Much better, thx

**Pull request merged**